nncf/torch/graph/patterns.py

```diff
@@ -11,27 +11,131 @@
  limitations under the License.
 """
 
-from nncf.common.graph import NNCFNodeExpression as N
+from nncf.common.graph.patterns import GraphPattern
+from nncf.common.utils.registry import Registry
 
-LINEAR_OPS = N('linear') | N('conv2d') | N('conv_transpose2d') | N('conv3d') | \
-             N('conv_transpose3d') | N('conv1d') | N('addmm')
+from nncf.torch.graph.pattern_operations import LINEAR_OPERATIONS
+from nncf.torch.graph.pattern_operations import BATCH_NORMALIZATION_OPERATIONS
+from nncf.torch.graph.pattern_operations import ACTIVATIONS_OPERATIONS
+from nncf.torch.graph.pattern_operations import ARITHMETIC_OPERATIONS
 
-RELU = N('relu') | N('relu_') | N('hardtanh')
+PATTERN_GRAPH = Registry('pattern_graph')
 
-BN = N('batch_norm') | N('batch_norm3d')
 
-POOLING = N('adaptive_avg_pool2d') | N('adaptive_avg_pool3d') | N('avg_pool2d') | N('avg_pool3d')
+def create_linear_swish_act() -> GraphPattern:
+    main_pattern = GraphPattern()
 
-NON_RELU_ACTIVATIONS = N('elu') | N('elu_') | N('prelu') | N('sigmoid') | N('gelu')
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ACTIVATIONS = RELU | NON_RELU_ACTIVATIONS
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ANY_BN_ACT_COMBO = BN + ACTIVATIONS | ACTIVATIONS + BN | BN | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-SINGLE_OPS = ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ARITHMETIC = N('__iadd__') | N('__add__') | N('__mul__') | N('__rmul__')
+    pattern.add_edge(linear_ops_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ELTWISE_UNIFORM_OPS = BN | RELU | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-MATMUL = N('matmul') | N('bmm')
+    return main_pattern
+
+
+def create_linear_h_swish_act() -> GraphPattern:
+    main_pattern = GraphPattern()
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
```

Comment:
Avoid double underscore in regular variable names.

Reply:
Changed names